src/framework/core/js/FluidRequests.js

```diff
@@ -343,5 +343,205 @@ var fluid_2_0 = fluid_2_0 || {};
+     * it is not limited to this method of interaction.
+     */
+    fluid.defaults("fluid.dataSource", {
+        gradeNames: ["fluid.eventedComponent", "autoInit"]
+        // Invokers should be defined for the typical CRUD functions and return
+        // a promise.
+        //
+        // The "get" and "delete" methods require the signature (directModel).
+        // The "set" method requires the signature (directModel, model)
+        //
+        // directModel: An object expressing an "index" into some set of
+        //              state which can be read or written.
+        //
+        // model: The payload sent to the storage.
+        //
+        // options: An object expressing implementation specific details
+        //          regarding the handling of a request. Note: this does not
+        //          include details for identifying the resource. Those should be
+        //          placed in the directModel.
+        //
+        // invokers: {
+        //     "get": {}, // handles the Read function
+        //     "set": {}, // handles the Create and Update functions
+        //     "delete": {} // handles the Delete function
+        // }
+    });
+
+    /*
+     * Converts an object or array to string for use as a key.
+     * The objects are sorted alphabetically to insure that they
+     * result in the same string across executions.
+     */
+    fluid.objectToHashKey = function (obj) {
+        var str = fluid.isArrayable(obj) ? "array " : "object ";
+        var keys = fluid.keys(obj).sort();
+
+        fluid.each(keys, function (key) {
+            var val = obj[key];
+            str += key + ":" + fluid.toHashKey(val);
+        });
+
+        return str;
+    };
+
+    /*
+     * Generates a string for use as a key.
+     * They typeof of the value passed in will be prepended
+     * to ensure that (strings vs numbers) and (arrays vs objects)
+     * are distinguishable.
+     */
+    fluid.toHashKey = function (val) {
+        var str;
+        if(fluid.isPlainObject(val)){
+            str = "<" + fluid.objectToHashKey(val) + ">";
+        } else {
+            str = "|" + JSON.stringify(val) + "|";
+        }
+        return str;
+    };
+
+    /*
+     * A dataSource wrapper providing a queuing mechanism for requests.
+     * Requests are queued based on type (read/write) and resource (directModel).
+     *
+     * A fully implemented dataSource, following the structure outlined by fluid.dataSource,
+     * must be provided in the wrappedDataSource subcomponent. The get, set, and delete methods
+     * found on the queuedDataSource will call their counterparts in the wrappedDataSource, after
+     * filtering through the appropriate queue.
+     *
+     * TODO: A fully realized implementation should provide a mechansim for working with a local
+     *       cache. For example pending write requests could be used to service get requests directly.
+     */
+    fluid.defaults("fluid.queuedDataSource", {
```

Comment:
created a new grade called debouncedDatasource that attaches the enqueue methods.